### ISSUE TITLE
App running in docker container can't connect to Ollama container

### ISSUE THREAD

**Issue by thomassrour** [ollama/ollama#3652]
I have a chainlit app running in a container, and another container running Ollama, but the first can't find the second. I saw some issues that described the same problem, but no matter what I try, I always get the error :

HTTPConnectionPool(host='localhost', port=11434): Max retries exceeded with url: /api/generate (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe894516620>: Failed to establish a new connection: [Errno 111] Connection refused'))

I can access the ollama container from my host machine just fine, the problem appears when I try accessing it from the other container. Here is my docker compose :

version: '3.8'

services:
  app:
    image: rag:latest
    ports:
      - "52000:52000"
    extra_hosts:
      - "host.docker.internal:host-gateway"
    environment:
      DATABASE_URL: ...
      OLLAMA_HOST: 0.0.0.0
    restart: always
    deploy:
      resources:
        reservations:
          devices:
          - driver: nvidia
            capabilities: [gpu]
            count: all

  db:
  #container_name: rag_db
    image: postgres:14-alpine
    environment:
      POSTGRES_DB: ...
      POSTGRES_USER: ...
      POSTGRES_PASSWORD: ...
    volumes:
      - pgdata:/var/lib/postgresql/data
    ports:
      - "52001:5432" 
    restart: always

  ollama-container:
    hostname: ollama-container
    image: ollama/ollama
    environment:
      OLLAMA_HOST: 0.0.0.0
    volumes:
      - ./data/ollama:/root/.ollama
    ports:
      - "11434:11434"
    extra_hosts:
      - "host.docker.internal:host-gateway"
    deploy:
      resources:
        reservations:
          devices:
          - driver: nvidia
            capabilities: [gpu]
            count: all

volumes:
  pgdata:

Any help would be much appreciated, thanks.

**Comment by thomassrour** [comment]
Hello, sure, here they are :

ollama-container_1  | time=2024-04-15T12:46:07.336Z level=INFO source=images.go:804 msg="total blobs: 5"
ollama-container_1  | time=2024-04-15T12:46:07.336Z level=INFO source=images.go:811 msg="total unused blobs removed: 0"
ollama-container_1  | time=2024-04-15T12:46:07.336Z level=INFO source=routes.go:1118 msg="Listening on [::]:11434 (version 0.1.30)"
ollama-container_1  | time=2024-04-15T12:46:07.337Z level=INFO source=payload_common.go:113 msg="Extracting dynamic libraries to /tmp/ollama818474866/runners ..."
ollama-container_1  | time=2024-04-15T12:46:11.447Z level=INFO source=payload_common.go:140 msg="Dynamic LLM libraries [cpu_avx cpu_avx2 cuda_v11 rocm_v60000 cpu]"
ollama-container_1  | time=2024-04-15T12:46:11.447Z level=INFO source=gpu.go:115 msg="Detecting GPU type"
ollama-container_1  | time=2024-04-15T12:46:11.447Z level=INFO source=gpu.go:265 msg="Searching for GPU management library libcudart.so*"
ollama-container_1  | time=2024-04-15T12:46:11.448Z level=INFO source=gpu.go:311 msg="Discovered GPU libraries: [/tmp/ollama818474866/runners/cuda_v11/libcudart.so.11.0]"
ollama-container_1  | time=2024-04-15T12:46:11.473Z level=INFO source=gpu.go:120 msg="Nvidia GPU detected via cudart"
ollama-container_1  | time=2024-04-15T12:46:11.473Z level=INFO source=cpu_common.go:11 msg="CPU has AVX2"
ollama-container_1  | time=2024-04-15T12:46:11.558Z level=INFO source=gpu.go:188 msg="[cudart] CUDART CUDA Compute Capability detected: 8.6"

**Comment by amila-ku** [comment]
@thomassrour Can you try to access by setting hostname as ollama-container(http://ollama-container:11434/api/generate) in your app.

**Comment by thomassrour** [comment]
@amila-ku It worked, thanks! For anyone else using langchain, 

`llm = Ollama(model="mixtral", verbose=False, temperature=0, base_url="http://ollama-container:11434")`

**Comment by ntelo007** [comment]
Hey guys. For me it's still persists:

My docker-compose.yaml:
```
services:
  ollama:
    image: ollama/ollama
    container_name: ollama
    volumes:
      - ./data/ollama:/root/.ollama
    ports:
      - 11434:11434
    environment:
      - OLLAMA_HOST=0.0.0.0
      - OLLAMA_ORIGINS=http://0.0.0.0:11434
    healthcheck:
      test: ["CMD", "ollama", "list"]
      interval: 10s
      timeout: 30s
      retries: 5
      start_period: 10s
    networks:
      - fastapi-app-network

  fastapi_app:
    image: fastapi_image
    ports:
      - 8501:8501
    container_name: fastapi_app
    depends_on:
      - ollama
    networks:
      - fastapi-app-network

volumes:
  ollama: {}

networks:
  fastapi-app-network:
    driver: bridge
```

This is how I create an llm object in langchain:
```
# /// LLM ///
llm = ChatOllama(
    model='llama3-groq-tool-use',  # or 'llama3.1'
    base_url='http://ollama:11434',
    temperature=0,
    verbose=True
)
```

Some logs:
```
2024-08-27 19:03:21 2024/08/27 17:03:21 routes.go:1123: INFO server config env="map[CUDA_VISIBLE_DEVICES: GPU_DEVICE_ORDINAL: HIP_VISIBLE_DEVICES: HSA_OVERRIDE_GFX_VERSION: OLLAMA_DEBUG:false OLLAMA_FLASH_ATTENTION:false OLLAMA_HOST:http://0.0.0.0:11434 OLLAMA_INTEL_GPU:false OLLAMA_KEEP_ALIVE:5m0s OLLAMA_LLM_LIBRARY: OLLAMA_MAX_LOADED_MODELS:0 OLLAMA_MAX_QUEUE:512 OLLAMA_MODELS:/root/.ollama/models OLLAMA_NOHISTORY:false OLLAMA_NOPRUNE:false OLLAMA_NUM_PARALLEL:0 OLLAMA_ORIGINS:[http://0.0.0.0:11434 http://localhost https://localhost http://localhost:* https://localhost:* http://127.0.0.1 https://127.0.0.1 http://127.0.0.1:* https://127.0.0.1:* http://0.0.0.0 https://0.0.0.0 http://0.0.0.0:* https://0.0.0.0:* app://* file://* tauri://*] OLLAMA_RUNNERS_DIR: OLLAMA_SCHED_SPREAD:false OLLAMA_TMPDIR: ROCR_VISIBLE_DEVICES:]"
2024-08-27 19:03:21 time=2024-08-27T17:03:21.689Z level=INFO source=images.go:782 msg="total blobs: 5"
2024-08-27 19:03:21 time=2024-08-27T17:03:21.724Z level=INFO source=images.go:790 msg="total unused blobs removed: 0"
2024-08-27 19:03:21 time=2024-08-27T17:03:21.764Z level=INFO source=routes.go:1170 msg="Listening on [::]:11434 (version 0.3.5)"
2024-08-27 19:03:21 time=2024-08-27T17:03:21.766Z level=INFO source=payload.go:30 msg="extracting embedded files" dir=/tmp/ollama556781795/runners
2024-08-27 19:03:26 time=2024-08-27T17:03:26.736Z level=INFO source=payload.go:44 msg="Dynamic LLM libraries [cpu_avx cpu_avx2 cuda_v11 rocm_v60102 cpu]"
2024-08-27 19:03:26 time=2024-08-27T17:03:26.736Z level=INFO source=gpu.go:204 msg="looking for compatible GPUs"
2024-08-27 19:03:26 time=2024-08-27T17:03:26.737Z level=INFO source=gpu.go:350 msg="no compatible GPUs were discovered"
2024-08-27 19:03:26 time=2024-08-27T17:03:26.737Z level=INFO source=types.go:105 msg="inference compute" id=0 library=cpu compute="" driver=0.0 name="" total="15.5 GiB" available="14.2 GiB"
2024-08-27 19:03:26 [GIN] 2024/08/27 - 17:03:26 | 200 |      28.902µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:03:26 [GIN] 2024/08/27 - 17:03:26 | 200 |   34.619766ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:03:36 [GIN] 2024/08/27 - 17:03:36 | 200 |      19.802µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:03:36 [GIN] 2024/08/27 - 17:03:36 | 200 |   39.705815ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:03:46 [GIN] 2024/08/27 - 17:03:46 | 200 |      20.201µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:03:46 [GIN] 2024/08/27 - 17:03:46 | 200 |   29.007176ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:03:56 [GIN] 2024/08/27 - 17:03:56 | 200 |      19.601µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:03:56 [GIN] 2024/08/27 - 17:03:56 | 200 |   30.969158ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:04:07 [GIN] 2024/08/27 - 17:04:07 | 200 |      50.004µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:04:07 [GIN] 2024/08/27 - 17:04:07 | 200 |   51.390499ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:04:17 [GIN] 2024/08/27 - 17:04:17 | 200 |      22.102µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:04:17 [GIN] 2024/08/27 - 17:04:17 | 200 |   28.691946ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:04:27 [GIN] 2024/08/27 - 17:04:27 | 200 |      35.603µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:04:27 [GIN] 2024/08/27 - 17:04:27 | 200 |    28.08129ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:04:37 [GIN] 2024/08/27 - 17:04:37 | 200 |      20.602µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:04:37 [GIN] 2024/08/27 - 17:04:37 | 200 |   29.417075ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:04:47 [GIN] 2024/08/27 - 17:04:47 | 200 |      33.502µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:04:47 [GIN] 2024/08/27 - 17:04:47 | 200 |   34.018837ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:04:57 [GIN] 2024/08/27 - 17:04:57 | 200 |      20.602µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:04:57 [GIN] 2024/08/27 - 17:04:57 | 200 |   29.486483ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:05:07 [GIN] 2024/08/27 - 17:05:07 | 200 |      36.003µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:05:07 [GIN] 2024/08/27 - 17:05:07 | 200 |   39.233568ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:05:17 [GIN] 2024/08/27 - 17:05:17 | 200 |      20.501µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:05:17 [GIN] 2024/08/27 - 17:05:17 | 200 |   30.003403ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:05:27 [GIN] 2024/08/27 - 17:05:27 | 200 |      20.001µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:05:27 [GIN] 2024/08/27 - 17:05:27 | 200 |   31.734826ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:05:37 [GIN] 2024/08/27 - 17:05:37 | 200 |      18.802µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:05:37 [GIN] 2024/08/27 - 17:05:37 | 200 |   27.485754ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:05:47 [GIN] 2024/08/27 - 17:05:47 | 200 |      19.102µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:05:47 [GIN] 2024/08/27 - 17:05:47 | 200 |   27.166987ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:05:57 [GIN] 2024/08/27 - 17:05:57 | 200 |      18.902µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:05:57 [GIN] 2024/08/27 - 17:05:57 | 200 |   27.496015ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:06:07 [GIN] 2024/08/27 - 17:06:07 | 200 |      20.001µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:06:07 [GIN] 2024/08/27 - 17:06:07 | 200 |   28.129247ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:06:17 [GIN] 2024/08/27 - 17:06:17 | 200 |      19.401µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:06:18 [GIN] 2024/08/27 - 17:06:18 | 200 |   27.624628ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:06:28 [GIN] 2024/08/27 - 17:06:28 | 200 |      36.303µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:06:28 [GIN] 2024/08/27 - 17:06:28 | 200 |   26.860463ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:06:38 [GIN] 2024/08/27 - 17:06:38 | 200 |      18.601µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:06:38 [GIN] 2024/08/27 - 17:06:38 | 200 |   26.443389ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:06:48 [GIN] 2024/08/27 - 17:06:48 | 200 |      19.002µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:06:48 [GIN] 2024/08/27 - 17:06:48 | 200 |   24.506714ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:06:58 [GIN] 2024/08/27 - 17:06:58 | 200 |      20.502µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:06:58 [GIN] 2024/08/27 - 17:06:58 | 200 |   29.411413ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:07:08 [GIN] 2024/08/27 - 17:07:08 | 200 |      21.202µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:07:08 [GIN] 2024/08/27 - 17:07:08 | 200 |    32.17346ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:07:18 [GIN] 2024/08/27 - 17:07:18 | 200 |      22.302µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:07:18 [GIN] 2024/08/27 - 17:07:18 | 200 |   27.878343ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:07:28 [GIN] 2024/08/27 - 17:07:28 | 200 |      20.602µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:07:28 [GIN] 2024/08/27 - 17:07:28 | 200 |   30.586642ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:07:38 [GIN] 2024/08/27 - 17:07:38 | 200 |      21.401µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:07:38 [GIN] 2024/08/27 - 17:07:38 | 200 |   35.459234ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:07:48 [GIN] 2024/08/27 - 17:07:48 | 200 |      21.702µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:07:48 [GIN] 2024/08/27 - 17:07:48 | 200 |   28.253535ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:07:58 [GIN] 2024/08/27 - 17:07:58 | 200 |      19.402µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:07:58 [GIN] 2024/08/27 - 17:07:58 | 200 |   28.828046ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:08:08 [GIN] 2024/08/27 - 17:08:08 | 200 |      27.002µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:08:08 [GIN] 2024/08/27 - 17:08:08 | 200 |   40.621965ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:08:18 [GIN] 2024/08/27 - 17:08:18 | 200 |      20.202µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:08:18 [GIN] 2024/08/27 - 17:08:18 | 200 |   28.376167ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:08:28 [GIN] 2024/08/27 - 17:08:28 | 200 |      18.701µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:08:28 [GIN] 2024/08/27 - 17:08:28 | 200 |   27.783714ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:08:38 [GIN] 2024/08/27 - 17:08:38 | 200 |      22.702µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:08:38 [GIN] 2024/08/27 - 17:08:38 | 200 |   31.336529ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:08:48 [GIN] 2024/08/27 - 17:08:48 | 200 |      20.002µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:08:49 [GIN] 2024/08/27 - 17:08:49 | 200 |   28.672512ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:08:59 [GIN] 2024/08/27 - 17:08:59 | 200 |      20.201µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:08:59 [GIN] 2024/08/27 - 17:08:59 | 200 |   28.218635ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:09:09 [GIN] 2024/08/27 - 17:09:09 | 200 |      20.201µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:09:09 [GIN] 2024/08/27 - 17:09:09 | 200 |   41.087411ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:09:19 [GIN] 2024/08/27 - 17:09:19 | 200 |      19.201µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:09:19 [GIN] 2024/08/27 - 17:09:19 | 200 |   27.341343ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:09:29 [GIN] 2024/08/27 - 17:09:29 | 200 |      18.302µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:09:29 [GIN] 2024/08/27 - 17:09:29 | 200 |   27.771992ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:09:39 [GIN] 2024/08/27 - 17:09:39 | 200 |      18.901µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:09:39 [GIN] 2024/08/27 - 17:09:39 | 200 |   27.004426ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:09:49 [GIN] 2024/08/27 - 17:09:49 | 200 |      21.902µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:09:49 [GIN] 2024/08/27 - 17:09:49 | 200 |   22.520824ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:09:59 [GIN] 2024/08/27 - 17:09:59 | 200 |      29.402µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:09:59 [GIN] 2024/08/27 - 17:09:59 | 200 |   22.370515ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:10:09 [GIN] 2024/08/27 - 17:10:09 | 200 |      18.501µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:10:09 [GIN] 2024/08/27 - 17:10:09 | 200 |   36.584739ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:10:19 [GIN] 2024/08/27 - 17:10:19 | 200 |      22.702µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:10:19 [GIN] 2024/08/27 - 17:10:19 | 200 |   26.692761ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:10:29 [GIN] 2024/08/27 - 17:10:29 | 200 |      19.401µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:10:29 [GIN] 2024/08/27 - 17:10:29 | 200 |    27.38842ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:10:39 [GIN] 2024/08/27 - 17:10:39 | 200 |      18.601µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:10:39 [GIN] 2024/08/27 - 17:10:39 | 200 |   29.575365ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:10:49 [GIN] 2024/08/27 - 17:10:49 | 200 |      20.401µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:10:49 [GIN] 2024/08/27 - 17:10:49 | 200 |   28.826732ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:10:59 [GIN] 2024/08/27 - 17:10:59 | 200 |      23.002µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:10:59 [GIN] 2024/08/27 - 17:10:59 | 200 |   27.384409ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:11:09 [GIN] 2024/08/27 - 17:11:09 | 200 |      20.202µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:11:10 [GIN] 2024/08/27 - 17:11:10 | 200 |   33.701944ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:11:20 [GIN] 2024/08/27 - 17:11:20 | 200 |      23.402µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:11:20 [GIN] 2024/08/27 - 17:11:20 | 200 |   28.152542ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:11:30 [GIN] 2024/08/27 - 17:11:30 | 200 |      22.902µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:11:30 [GIN] 2024/08/27 - 17:11:30 | 200 |   29.095507ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:11:40 [GIN] 2024/08/27 - 17:11:40 | 200 |      19.601µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:11:40 [GIN] 2024/08/27 - 17:11:40 | 200 |   29.740486ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:11:50 [GIN] 2024/08/27 - 17:11:50 | 200 |      24.801µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:11:50 [GIN] 2024/08/27 - 17:11:50 | 200 |    35.84487ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:12:00 [GIN] 2024/08/27 - 17:12:00 | 200 |      22.202µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:12:00 [GIN] 2024/08/27 - 17:12:00 | 200 |   27.348868ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:12:10 [GIN] 2024/08/27 - 17:12:10 | 200 |      35.503µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:12:10 [GIN] 2024/08/27 - 17:12:10 | 200 |   35.850264ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:12:20 [GIN] 2024/08/27 - 17:12:20 | 200 |      20.401µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:12:20 [GIN] 2024/08/27 - 17:12:20 | 200 |   38.821102ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:12:30 [GIN] 2024/08/27 - 17:12:30 | 200 |      20.402µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:12:30 [GIN] 2024/08/27 - 17:12:30 | 200 |   27.638242ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:12:40 [GIN] 2024/08/27 - 17:12:40 | 200 |      20.401µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:12:40 [GIN] 2024/08/27 - 17:12:40 | 200 |   29.076272ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:12:50 [GIN] 2024/08/27 - 17:12:50 | 200 |      18.202µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:12:50 [GIN] 2024/08/27 - 17:12:50 | 200 |   30.331616ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:13:00 [GIN] 2024/08/27 - 17:13:00 | 200 |      18.401µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:13:00 [GIN] 2024/08/27 - 17:13:00 | 200 |   26.408142ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:13:10 [GIN] 2024/08/27 - 17:13:10 | 200 |      21.902µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:13:11 [GIN] 2024/08/27 - 17:13:11 | 200 |    29.09343ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:13:21 [GIN] 2024/08/27 - 17:13:21 | 200 |      22.001µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:13:21 [GIN] 2024/08/27 - 17:13:21 | 200 |   26.770545ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:13:31 [GIN] 2024/08/27 - 17:13:31 | 200 |     177.412µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:13:31 [GIN] 2024/08/27 - 17:13:31 | 200 |   27.866636ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:13:41 [GIN] 2024/08/27 - 17:13:41 | 200 |      21.801µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:13:41 [GIN] 2024/08/27 - 17:13:41 | 200 |   28.738729ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:13:51 [GIN] 2024/08/27 - 17:13:51 | 200 |      21.702µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:13:51 [GIN] 2024/08/27 - 17:13:51 | 200 |   30.152366ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:14:01 [GIN] 2024/08/27 - 17:14:01 | 200 |      19.501µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:14:01 [GIN] 2024/08/27 - 17:14:01 | 200 |   27.556787ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:14:11 [GIN] 2024/08/27 - 17:14:11 | 200 |      29.902µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:14:11 [GIN] 2024/08/27 - 17:14:11 | 200 |   29.282004ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:14:21 [GIN] 2024/08/27 - 17:14:21 | 200 |      21.202µs |       127.0.0.1 | HEAD     "/"
2024-08-27 19:14:21 [GIN] 2024/08/27 - 17:14:21 | 200 |   29.772453ms |       127.0.0.1 | GET      "/api/tags"
2024-08-27 19:14:31 [GIN] 2024/08/27 - 17:14:31 | 200 |      20.101µs |       127.0.0.1 | HEAD     "/"
```
